vendor/engines/bulk_email/app/services/bulk_email/recipient_searcher.rb

```diff
@@ -62,6 +62,14 @@ def search_authorized_users
         .reorder(*self.class::DEFAULT_SORT)
     end
 
+    def search_training_requested
+      users
+        .joins(:training_requests)
+        .uniq
+        .where(training_requests: { product_id: product_ids_from_params })
```

Comment:
I think we handle it, but can we make sure this does not include users who issued a training request, but then were later approved? We actually destroy training requests on the "destroy" action, so this shouldn't include users who's request was denied.